src/PackageExtensionCompat.jl

```diff
@@ -5,39 +5,55 @@ export @require_extensions
 const HAS_NATIVE_EXTENSIONS = isdefined(Base, :get_extension)
 
 @static if !HAS_NATIVE_EXTENSIONS
-    using Requires, TOML
+    using MacroTools, Requires, TOML
 
-    function rewrite_import(str, pkgs)
-        parts = split(strip(str))
-        if length(parts) == 1 || (length(parts) ≥ 2 && parts[2] == "as")
-            if parts[1] ∈ pkgs
-                parts[1] = string("..", parts[1])
+    @static if hasmethod(Base.include, Tuple{Function, Module, String})
+        function _include(mapexpr::Function, m::Module, path::AbstractString)
+            Base.include(mapexpr, m, path)
+        end
+    else
+        function _include(mapexpr::Function, m::Module, path::AbstractString)
+            path = abspath(path)
+            cd(dirname(path)) do
+                str = read(basename(path), String)
+                pos = 1
+                while true
```

Comment:
The file should contain a single expression (the module implementing the extension) so you don't need a loop - can just `expr = Meta.parse(str)`.

Reply:
For the extension module, I agree that it should. Is there anything, though, that prevents the developer to put more than one block in a file? Also, this same `_include` function will be used to load additional files, e.g., files implementing parts of the extension module, and it is perfectly legal to have multiple definitions in those files. So I think that the for loop cannot be omitted.

Reply:
I assume an exception is thrown if the top level file doesn't contain exactly a single `module WeakDep ... end` expression. And I don't think you actually use `_import` anywhere but the top level? Any imports in the extension code are left as-is.

Reply:
Still if it works it works and I'm happy to leave it.